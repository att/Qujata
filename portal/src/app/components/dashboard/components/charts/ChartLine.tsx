/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/typedef */
/* eslint-disable jsx-a11y/label-has-associated-control */
import { ChartData, ChartOptions } from 'chart.js';
import React, { } from 'react';
import { Line } from 'react-chartjs-2';

import styles from './ChartLine.module.scss';
import { ChartLineColors, ChartOption } from './Charts.const';

export interface ChartLineProps {
  title: string;
  data: Map<string, number[]>
}

export const ChartLine: React.FC<ChartLineProps> = (props: ChartLineProps) => {
  const { title: name, data: linesData } = props;
  const datasets: { label: string, data: number[], fill: boolean, borderColor: string }[] = [];
  let lineIndex:number = 0;
  linesData.forEach((value, key) => {
    datasets.push({ label: key, data: value, fill: true, borderColor: ChartLineColors[lineIndex] });
    lineIndex += 1;
  });

  const config: ChartOptions<'line'> = {
    ...ChartOption,
    plugins: {
      ...ChartOption.plugins, legend: { position: 'top', display: true },
    },
  };

  const data: ChartData<'line'> = { labels: Array.from(Array(datasets[0].data.length).keys()), datasets };

  return (
    <div className={styles.wrapper}>
      <div className={styles.name}>{name}</div>
      <Line data={data} options={config} />
    </div>
  );
};
