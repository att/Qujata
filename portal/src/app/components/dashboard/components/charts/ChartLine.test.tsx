import '@testing-library/jest-dom';
import { RenderResult, render, screen } from '@testing-library/react';
import { ChartLine, ChartLineProps } from './ChartLine';

describe('ChartLine', () => {
  let props: ChartLineProps;
  beforeAll(() => {
    props = {
      title: 'Sample Title',
      data: new Map([
        ['Line 1', [1, 2, 3]],
        ['Line 2', [4, 5, 6]],
      ]),
    };
  });

  test('should render ChartLine', () => {
    // Render the ChartLine component with the prepared props
    const { container }: RenderResult = render(<ChartLine {...props} />);

    // Snapshot Matching
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should check all values were rendered', () => {
    // Render the ChartLine component with the prepared props
    render(<ChartLine {...props} />);

    // Assert that the component title is rendered
    expect(screen.getByText('Sample Title')).toBeInTheDocument();

    // Assert that the chart is rendered
    expect(screen.getByRole('img')).toBeInTheDocument();
  });
});
