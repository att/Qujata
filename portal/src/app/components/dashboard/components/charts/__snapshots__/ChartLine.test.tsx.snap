// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartLine should render ChartLine 1`] = `
<div
  class="wrapper"
>
  <div
    class="name"
  >
    Sample Title
  </div>
  <canvas
    height="150"
    role="img"
    width="300"
  />
</div>
`;
